mailet/standard/src/main/java/org/apache/james/transport/mailets/ContactExtractor.java

```diff
@@ -93,13 +93,24 @@ public void service(Mail mail) throws MessagingException {
     }
 
     private Optional<String> extractContacts(MimeMessage mimeMessage) throws MessagingException {
-        return Optional.ofNullable(mimeMessage.getSender())
+        return getSender(mimeMessage)
                 .map(Address::toString)
                 .filter(Throwing.predicate(sender -> hasRecipients(mimeMessage)))
                 .map(Throwing.function(sender -> new ExtractedContacts(sender, recipients(mimeMessage))))
                 .map(Throwing.function(message -> objectMapper.writeValueAsString(message)));
     }
 
+    private Optional<Address> getSender(MimeMessage mimeMessage) throws MessagingException {
+        Optional<Address> sender = Optional.ofNullable(mimeMessage.getSender());
+        if (sender.isPresent()) {
+            return sender;
+        } else {
+            return Optional.ofNullable(mimeMessage.getFrom())
+                .filter(addresses -> addresses.length > 0)
+                .map(addresses -> addresses[0]);
+        }
```

Comment:
so if I send a message without a sender and with a forged From, I can add contacts to an other user?

Reply:
Ah... That's not stupid.

Maybe we need **TO GET SENDER FROM THE ENVELOPE** but recipients from the Mime Message